README.md

```diff
@@ -1,6 +1,302 @@
 Rancher API Server
 ==================
 
-This repo contains the low level API server framework used to create REST APIs consumed by Rancher projects such as
-github.com/rancher/ui and github.com/rancher/dashboard.  The primary usage of this framework is by github.com/rancher/steve
-and probably does not have general applicability beyond that.
\ No newline at end of file
+This repo contains the low level API server framework used to create REST APIs
+consumed by Rancher projects such as
+[github.com/rancher/ui](https://github.com/rancher/ui) and
+[github.com/rancher/dashboard](https://github.com/rancher/dashboard). The
+primary consumer of this framework is
+[github.com/rancher/steve](https://github.com/rancher/steve).
+
+Overview
+--------
+
+The API server is the interface between an HTTP client and a more complex
+application like [rancher/steve](https://github.com/rancher/steve). The two
+main components that are used to accomplish that are Schemas and Stores.
+
+Schemas define metadata about an API type, describe CRUD handlers for the type,
+define formatting transformations, and declare the Store that will be used to
+transform and store the object.
+
+Stores provide a common interface to perform CRUD operations on objects. The
+implementation of the interface commonly involves either storing the data as a
+field on the store object, forwarding it to another nested store, or calling
+out to an external resource like Kubernetes.
```

Comment:
From my understanding it more like:
```suggestion
implementation of the interface commonly involves either storing the data as a
field on the store object, then forwarding it to another nested store or calling
out to an external resource like Kubernetes.
```
I also think calling out to kubernetes is implemented as yet another nested store but maybe that detail is okay to be omitted.

Reply:
Oh okay that totally makes sense, thank you.